### DISCUSSION TITLE
How to debug discrepancies between CI and local

### DISCUSSION THREAD

**Discussion by georgiemathews** [vercel/turborepo#8809] [Help]
### Summary

Hi, would love some pointers on how I can debug discrepancies between the task graph when I try running a task locally and I try running in CI.

I'm using turbo 2.0.8, my local machine is an M1 mac and my CI is a Github Actions job running on Ubuntu. Neither have any special environment variables set.

The task I'm trying to run is defined with this in the root `turbo.json`:
```json
{
  "tasks": {
    "detox:build": {
      "dependsOn": ["^build"]
    },
    "detox:test": {
      "dependsOn": ["build"]
    }
  }
}
```

When running the `detox:test` task locally filtered on `mypackage`, I get the correct build graph:

```
yarn detox detox:test --filter='mypackage' --graph

digraph {
	compound = "true"
	newrank = "true"
	subgraph "root" {
		"[root] dep1#build" -> "[root] ___ROOT___"
		"[root] dep2#build" -> "[root] ___ROOT___"
		"[root] mypackage#detox:build" -> "[root] dep1#build"
		"[root] mypackage#detox:build" -> "[root] dep2#build"
		"[root] mypackage#detox:test" -> "[root] mypackage#build"
	}
}
```

When the same command runs in CI, it produces the incorrect output (doesn't include the appropriate dependent tasks):
```
yarn turbo detox:test --filter="mypackage" --graph
digraph {
	compound = "true"
	newrank = "true"
	subgraph "root" {
		"[root] mypackage#detox:test" -> "[root] ___ROOT___"
	}
}
```

I've scoured docs and github issues to try to find any info I'm missing about running in CI or Github Actions specifically, but coming up empty. Any ideas on how I can continue to debug this issue?

**Comment by georgiemathews** [comment]
@anthonyshew wondering if you have any pointers?

**Reply by anthonyshew** [reply]
🤔 This sounds like a bug rather than user error. If you could open an issue with a reproduction, that would be great.
